### ISSUE TITLE
Update media item playback info / Modify IndexedAudioSource Tag property

### ISSUE THREAD

**Issue by markst** [ryanheise/just_audio#610]
**Is your feature request related to a problem? Please describe.**
Our Radio app plays live audio with a schedule which updates ever hour or two.
We wish to update the meta & notification banner with updated schedule.

**Describe the solution you'd like**
Possible to update `audio_service` now playing info without having to update the source entirely.

**Describe alternatives you've considered**
Updating the source with new meta data on the change of schedule.

**Additional context**
![Screenshot_1639574993](https://user-images.githubusercontent.com/274318/146195464-f246425d-ff26-485a-a8f1-cc167870c84e.png)

**Comment by ryanheise** [comment]
I suppose that could be done, although I am not sure at the moment the best approach to take and I wouldn't implement it until that is clear.

In the meantime, if you have more advanced requirements, you are best off using audio_service directly since you have complete flexibility over what to display and when.

**Comment by markst** [comment]
Thanks for response @ryanheise.

Currently `just_audio_background` is fulfilling our needs without the need to use `audio_service` directly. However perhaps our best solution is to do so.

I'd need to dive deeper into `audio_service` to be sure we can update the media item independently of the audio source.

**Comment by dpacchi** [comment]
Hi,

I'm developing a radio app with multiple radio channels (streams) and trying to display the channel info in the lock screen.
I'm using the just_audio and just_audio_background packages.

When the user load or change channel this code is invoked:  

`await player.setAudioSource(
        AudioSource.uri(
          Uri.parse(channel.url ?? ""),
          tag: MediaItem(
            // Specify a unique ID for each media item:
            id: channel.id.toString(),
            // Metadata to display in the notification:
            album: null,
            title: channel.name ?? "",
            artUri: Uri.parse(channel.image ?? ""),
          ),
        ),
      );
      player.play();`

The first time the lock screen updates correctly with the selected channel info, then even if I change channel and replace the audio source the info in the lock screen is not updated and it keeps displaying the first mediaitem info.
The app UI is updated without problems tough. 

Also it would be great for me to display the current song info in the lock screen that I get with:

`  void metadataListner() {
    player.icyMetadataStream.listen((icyMetadata) {
      if (icyMetadata != null && icyMetadata.info != null) {
        var songMetadata = SongMetadata(icyMetadata);
        verboseLog("Now playing ${songMetadata.artist} - ${songMetadata.song}");
        emit(PlayerChannelLoaded(state.currentChannel, songMetadata));
      }
    });
  }`

Thanks!

**Comment by markst** [comment]
Hey @dpacchi. I've also experienced this same issue. But yet to reproduce or figure out why.

Can you confirm you're always providing a `MediaItem` with all attributes present? - even if empty string.

**Comment by dpacchi** [comment]
of course always filled with at least title = channel.name and artUri = channel.image

**Comment by ryanheise** [comment]
> I'd need to dive deeper into `audio_service` to be sure we can update the media item independently of the audio source.

Yes, audio_service allows you to do that because audio_service doesn't care about audio sources at all, it only cares about media items. So you just tell audio_service which media item to display at which time.

**Comment by markst** [comment]
I've attempted using `audio_service`. However the `_audioHandler.playMediaItem` doesn't appear to be working.

Using `_audioHandler.playFromUri` works fine, but doesn't display any of my meta data in now playing notification card.

**Comment by ryanheise** [comment]
Hi @markst I'm afraid I can't give a tutorial here in the comments section on how to use that package as it would not be the appropriate place to do so (nor would I have the time to help you in that particular way). However, I would suggest going through the various tutorials online, looking at the examples, and reading the FAQ, to understand how to use it. Then if you still have questions, StackOverflow is the place to go to for community help.

**Comment by markst** [comment]
Thanks Ryan, nor do I expect you to.

Seems the issue was due to `playMediaItem` not being implemented by the `just_audio_handlers`: https://github.com/yringler/inside-app/blob/master/just_audio_handlers/lib/src/handler_just_audio.dart

I'll have to made some modifications to the handler or make up my own.

**Comment by markst** [comment]
I'm able to update my lock screen banner using the following:

```dart
Future<void> updateMeta(MediaItem tag) async {
  return _audioHandler?.updateMediaItem(tag);
}
```
